main.tf

```diff
@@ -1,10 +1,179 @@
+locals {
+  name                 = "ondat"
+  service_account_name = "storageos-operator"
+
+  ondat_etcd_endpoints = length(var.etcd_endpoints) == 0 ? "storageos-etcd.storageos-etcd:2379" : join(",", var.etcd_endpoints)
+
+  argocd_gitops_config = {
+    enable                             = true
+    etcdClusterCreate                  = length(var.etcd_endpoints) == 0
+    serviceAccountName                 = local.service_account_name
+    clusterSecretRefName               = "storageos-api"
+    clusterAdminUsername               = "storageos"
+    clusterAdminPassword               = "storageos"
+    clusterKvBackendAddress            = local.ondat_etcd_endpoints
+    clusterKvBackendTLSSecretName      = length(kubernetes_secret.etcd) > 0 ? kubernetes_secret.etcd[0].metadata[0].name : "storageos-etcd-secret"
```

Comment:
I don't follow this logic - it seems like you need a `var.create_<something>_namespace` flag. Same for line below